doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -195,19 +285,548 @@ Mail chapel-developers@ for review and merge
+
+Note: This should only ever be done to commits that **have not been pushed** to
+a remote repository.
+
+.. _Read commit messages:
+
+Read commit messages for changes from the main Chapel project
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+To view only the commits that happened on master (in other words, the old svn
+commits and the merge commits for pull requests):
+
+.. code-block:: bash
+
+    git log --first-parent
+
+    # or with line wrapping
+    git log --first-parent | less
+
+    # or including files changed
+    git log --first-parent -m --stat
+
+    # or similar to svn log
+    git log --first-parent -m --name-status
+
+More logging commands are described in `Other logging commands`_ below.
+
+.. _How to push:
+
+How to push
++++++++++++
+
+.. code-block:: bash
+
+    git push origin <branch_name>
+
+    # or if you don't like typing your complicated branch name,
+    # you can use this command to push the current branch:
+    git push origin HEAD
+
+    # if you forgot your branch name, you can get it by running
+    git branch
+
+    # it is the starred one...
+
+.. _How to open a PR:
+
+How to open a PR:
++++++++++++++++++
+
+* `Submit a pull request`_ with your changes (make sure you have `synced with
+  the main repo`_).
+
+  To do this, after pushing your changes to your feature branch on GitHub,
+  you can use the GitHub web interface to create a pull request. Visit
+
+  ``https://github.com/<username>/chapel``
+
+  and look for a "Compare & pull request" button for your feature branch.
+  Alternatively, navigate to your feature branch, and click the green icon next
+  to the branch dropdown to "Compare, review, create a pull request".
+
+  Next, put in a message to your reviewer about the purpose of your pull request
+  and give the pull request a useful title.  Your PR message will introduce the
+  changes to reviewers and form the basis for the merge message.  See
+  `Final merge message`_ for recommendations on what that commit message should
+  look like.
+
+  You will have to have signed a contributors agreement.  See https://github.com/chapel-lang/chapel/tree/master/doc/rst/developer/contributorAgreements/
+
+  Your pull request will be available at a URL like:
+
+  ``https://github.com/chapel-lang/chapel/pull/<number>``
+
+  and you can discuss the patch with your reviewers there.
+
+.. _How to merge a PR:
+
+How to merge a PR:
+++++++++++++++++++
+
+If you have commit privileges (see `Who has commit access`_), navigate to the
+pull request:
+
+go to
+
+https://github.com/chapel-lang/chapel/pulls
+
+or
+
+``https://github.com/chapel-lang/chapel/pull/<number>``
+
+and click the friendly green button "Merge pull request" (it is possible to
+merge the pull request from the command line also and the pull request page has
+details). When you click "Merge pull request", you will need to enter a commit
+message. See `Final merge message`_ for a reminder on what that commit message
+should entail.
```

Comment:
The link here is good, but the text doesn't really say that we're generally copying the PR message in to the merge message box.